File: SeventySix.Server/SeventySix/Logging/Constants/LogLevelConstants.cs
// <copyright file="LogLevelConstants.cs" company="SeventySix">
// Copyright (c) SeventySix. All rights reserved.
// </copyright>

namespace SeventySix.Logging;

/// <summary>
/// Constants for Serilog log level strings used in API requests and responses.
/// These match Serilog's LogEventLevel enum string representations.
/// </summary>
public static class LogLevelConstants
{
	/// <summary>Verbose log level (most detailed, for tracing).</summary>
	public const string Verbose = "Verbose";

	/// <summary>Debug log level (internal system events).</summary>
	public const string Debug = "Debug";

	/// <summary>Information log level (normal operation events).</summary>
	public const string Information = "Information";

	/// <summary>Warning log level (abnormal but handled events).</summary>
	public const string Warning = "Warning";

	/// <summary>Error log level (functionality unavailable or unexpected).</summary>
	public const string Error = "Error";

	/// <summary>Fatal log level (application crash or critical failure).</summary>
	public const string Fatal = "Fatal";

	/// <summary>Critical log level (alias for Fatal, .NET-style naming).</summary>
	public const string Critical = "Critical";

	/// <summary>Valid log levels for query filtering (Serilog levels only).</summary>
	public static readonly string[] QueryLevels =
		[Verbose, Debug, Information, Warning, Error, Fatal];

	/// <summary>Valid log levels for creating logs (includes Critical alias).</summary>
	public static readonly string[] CreateLevels =
		[Verbose, Debug, Information, Warning, Error, Fatal, Critical];
}

File: SeventySix.Client/src/app/features/admin/logs/constants/log-level.constants.ts
// <copyright file="log-level.constants.ts" company="SeventySix">
// Copyright (c) SeventySix. All rights reserved.
// </copyright>

/**
 * Log level constants matching server Serilog levels.
 */

/** Verbose log level (most detailed). */
export const LOG_LEVEL_VERBOSE: string = "Verbose";

/** Debug log level. */
export const LOG_LEVEL_DEBUG: string = "Debug";

/** Information log level. */
export const LOG_LEVEL_INFORMATION: string = "Information";

/** Warning log level. */
export const LOG_LEVEL_WARNING: string = "Warning";

/** Error log level. */
export const LOG_LEVEL_ERROR: string = "Error";

/** Fatal log level. */
export const LOG_LEVEL_FATAL: string = "Fatal";

/** Critical log level (alias for Fatal). */
export const LOG_LEVEL_CRITICAL: string = "Critical";

/** Valid log levels for filtering. */
export const VALID_LOG_LEVELS: readonly string[] =
	[
	LOG_LEVEL_VERBOSE,
	LOG_LEVEL_DEBUG,
	LOG_LEVEL_INFORMATION,
	LOG_LEVEL_WARNING,
	LOG_LEVEL_ERROR,
	LOG_LEVEL_FATAL
] as const;
